### ISSUE TITLE
Issues #4: CADL Generated Code Issue in EventGrid SDK

### ISSUE THREAD

**Issue by sarangan12** [Azure/autorest.typescript#1821]
While working on the EventGrid SDK, generated from CADL files, I have noticed the following. The linter is reporting several errors for the following error: `@typescript-eslint/naming-convention`.

In addition to that, there is another linter issue also. The method documentation is in the following formar:

```
/**
 * Initialize a new instance of `AzureMessagingEventGridClient`
 * @param endpoint type: string, The host name of the namespace, e.g. namespaceName1.westus-1.eventgrid.azure.net
 * @param credentials type: TokenCredential|KeyCredential, uniquely identify client credential
 * @param options type: ClientOptions, the parameter for all optional parameters
 */
```

This format causes issue with the linter and reports error `tsdoc-param-tag-missing-hyphen` for the rule `tsdoc/syntax`. It should be in the following format:

```
/**
 * Initialize a new instance of `AzureMessagingEventGridClient`
 * @param endpoint - The host name of the namespace, e.g. namespaceName1.westus-1.eventgrid.azure.net
 * @param credentials - uniquely identify client credential
 * @param options - the parameter for all optional parameters
 */
```

Please fix the above issues. 

@joheredi @xirzec FYI.....

**Comment by qiaozha** [comment]
@joheredi @sarangan12 @xirzec   do you think we should provide type hint in the description like this document mentioned https://jsdoc.app/tags-param.html ？
